client/src/views/AddTask.vue

```diff
@@ -134,54 +149,201 @@ let task: Task = {
 };
 export default Vue.extend({
   name: "AddTask",
-  data(): { task: Task; tooltip: string } {
+  data(): {
+    code: string;
+    config: string;
+    codeChecks: Array<string>;
+    codeErrors: Array<string>;
+    configChecks1: Array<Array<string>>;
+    configChecks2: Array<Array<string>>;
+    configErrors: Array<string>;
+    task: Task;
+    tooltip: string;
+  } {
     return {
+      code: "",
+      config: `{
+  "START": 0,
+  "END": 10,
+  "BATCH_SIZE": 1,
+  "RESULT": "sum",
+  "PUBLIC_RESULT": false,
+  "ALLOW_ANONYMOUS_USERS": true
+}`,
+      codeChecks: [
+        "window",
+        "document",
+        "console",
+        "alert",
+        "prompt",
+        "confirm",
+        "eval",
+        "import",
+        "export",
+        "XMLHttpRequest",
+        "fetch"
+      ],
+      codeErrors: [],
+      configChecks1: [
+        ["START", "number"],
+        ["END", "number"],
+        ["BATCH_SIZE", "number"],
+        ["RESULT", "string"],
+        ["PUBLIC_RESULT", "boolean"]
+      ],
+      configChecks2: [["ALLOW_ANONYMOUS_USERS", "boolean"]],
+      configErrors: [],
       task,
       tooltip: `interface Config {
   START: number;
   END: number;
   BATCH_SIZE: number;
   RESULT: string;
-  PUBLIC_RESULT?: boolean;
+  PUBLIC_RESULT: boolean;
   ALLOW_ANONYMOUS_USERS?: boolean;
 }`
     };
   },
   mounted() {
-    this.task.config = `{
-  "START": 0,
-  "END": 10,
-  "BATCH_SIZE": 1,
-  "RESULT": "sum",
-  "PUBLIC_RESULT": false,
-  "ALLOW_ANONYMOUS_USERS": true
-}`;
+    this.task.config = this.config;
+  },
+  watch: {
+    code: function (): void {
+      this.codeErrors = [];
+      for (let i = 0; i < this.codeChecks.length; i++) {
+        if (new RegExp(this.codeChecks[i]).test(this.code)) {
+          this.codeErrors.push("Hey, " + this.codeChecks[i] + " is forbidden!");
+        }
+      }

```

Comment:
Maybe use .match to simplify the code?
```suggestion
      const regex: RegExp = /window|document|console|alert|prompt|confirm|eval|import|export|XMLHttpRequest|fetch/g;
      const found: Array<string> = regex.match(this.code);
      for (let i = 0; i < found.length; i++) {
        this.codeErrors.push("Hey, " + found[i] + " is forbidden!");
      }
```